### ISSUE TITLE
[NPCs] Navigation Meshes from Left 4 Dead

### ISSUE THREAD

**Issue by xCommandoBoocex** [mapbase-source/source-sdk-2013#240]
Mapbase already has a lot of features from the Left 4 Dead (and later) engine branches, such as VScript and the Game Instructor, so why not add navigation meshes too?

**Comment by Yankumo-Koishi** [comment]
It's basically impossible because nextbot/nav mesh isn't open source

**Comment by arbabf** [comment]
> It's basically impossible because nextbot/nav mesh isn't open source
This. Additionally, Nextbots/navmeshes are for multiplayer Source games, not singleplayer as the tag you've put on your issue seems to imply.

**Comment by xCommandoBoocex** [comment]
> It's basically impossible because nextbot/nav mesh isn't open source

Doesn't the alien swarm code include nextbot code? I mean, it reuses L4D code for lots of things - game instructor, navmesh, VScript

**Comment by Yankumo-Koishi** [comment]
> Doesn't the alien swarm code include nextbot code? I mean, it reuses L4D code for lots of things - game instructor, navmesh, VScript
Yeah it doesn't in any public source code, I mean we don't count leaked code because Mapbase won't accept any commits based on it

**Comment by Matty-64** [comment]
> It's basically impossible because nextbot/nav mesh isn't open source

Navigation mesh code is included in both Source 2013 SDKs and Alien Swarm's SDK.  It is located inside the server project, see the **nav_** files inside the **game/server** folder.

In order to build the code files for it, you must have **USE_NAV_MESH** preprocessor defined in your server project VPC, IIRC.

Getting NPCs to use the NavMesh instead of A.I. Nodes would be work, but it's not impossible.  No More Room in Hell achieved this with their zombie NPCs.

**Comment by Blixibon** [comment]
While navmesh code is indeed included with Source 2013, I'm not really sure what could be done with it within Mapbase. Getting HL2 NPCs to be capable of using it seems like a daunting task, and one which doesn't seem necessary to me.

If there is a specific use case for it, I would recommend opening a separate issue.
